_field-types/flat-object.md

```diff
@@ -0,0 +1,218 @@
+---
+layout: default
+title: Flat object
+nav_order: 43
+has_children: false
+parent: Object field types
+grand_parent: Supported field types
+---
+
+# Flat object field type
+
+In OpenSearch, you don't have to specify a mapping before indexing documents. If you don't specify a mapping, OpenSearch uses [dynamic mapping]({{site.url}}{{site.baseurl}}/field-types/mappings#dynamic-mapping) to map every field and its subfields in the document automatically. When you ingest documents such as logs, you may not know every field's subfield name and type in advance. In this case, dynamically mapping all new subfields can quickly lead to a "mapping explosion", where the growing number of fields may degrade the performance of your cluster. 
```

Comment:
not a blocker for this documentation, but it would be good for us to have more documentation on our site to define "mapping explosion." It is a specific issue where too many distinct fields cause issues on the cluster manager. Does that fit into the current doc site somewhere?

Reply:
@macohen I'm open to having a separate page if we have more information to put on it. If it's only this one definition, then I think the last sentence in this paragraph conveys the same information.